### PR TITLE
SNOW-1570717: Add documentation pages for DatetimeIndex

### PR DESCRIPTION
**Pull request by sfc-gh-nkumar**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1570717

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Add documentation pages for DatetimeIndex. Add methods and attributes to indexing.rst. Also add datetime_index_supported.rst

### CODE REVIEW

**Review by sfc-gh-azhan — APPROVED** [review]
Thanks!

**Review by sfc-gh-evandenberg — APPROVED** [review]
LGTM

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM. Thanks, Naresh!

File: docs/source/modin/supported/datetime_index_supported.rst [hunk 1]
```diff
@@ -0,0 +1,284 @@
+``pd.DatetimeIndex`` supported APIs
```
Comment (sfc-gh-helmeleegy) [inline docs/source/modin/supported/datetime_index_supported.rst:1]: Actually, after a second look, can we limit the list of attributes and methods here to the ones listed in the [pandas docs for DatetimeIndex](https://pandas.pydata.org/docs/reference/api/pandas.DatetimeIndex.html). Many of the currently listed ones here belong to `Index` but not `DatetimeIndex`.

**Review by sfc-gh-helmeleegy — CHANGES_REQUESTED** [review]
I'm taking back my approval for now - as I think we first need to sync our list of attributes/methods with the pandas docs.

File: docs/source/modin/supported/datetime_index_supported.rst (hunk 1 above)
Comment (sfc-gh-nkumar) [reply docs/source/modin/supported/datetime_index_supported.rst:1]: done

**Review by sfc-gh-helmeleegy — APPROVED** [review]

**Pull request merged**